File: bindings/python/examples/df.py
import os, sigar;

sg = sigar.open()
fslist = sg.file_system_list()

def format_size(size):
    return sigar.format_size(size * 1024)

print 'Filesystem\tSize\tUsed\tAvail\tUse%\tMounted on\tType\n'

for fs in fslist:
    dir_name = fs.dir_name()
    usage = sg.file_system_usage(dir_name)

    total = usage.total()
    used = total - usage.free()
    avail = usage.avail()
    pct = usage.use_percent() * 100
    if pct == 0.0:
        pct = '-'

    print fs.dev_name(), format_size(total), format_size(used), format_size(avail),\
        pct, dir_name, fs.sys_type_name(), '/', fs.type_name()

File: bindings/perl/examples/df.pl
#!/usr/bin/perl

use strict;
use Sigar;

my $sigar = new Sigar;

my $fslist = $sigar->file_system_list;

print "Filesystem\tSize\tUsed\tAvail\tUse%\tMounted on\tType\n";

for my $fs (@$fslist) {
    my $dirname = $fs->dir_name;
    my $usage;

    #e.g. on win32 D:\ fails with "Device not ready"
    #if there is no cd in the drive.
    eval {
        $usage = $sigar->file_system_usage($dirname);
    } or next;

    my $total = $usage->total;
    my $used  = $total - $usage->free;
    my $avail = $usage->avail;
    my $pct   = $usage->use_percent * 100;
    my $usePct;

    if ($pct == 0) {
        $usePct = "-"
    }
    else {
        $usePct = $pct . '%';
    }

    print
      $fs->dev_name . "\t" .
      format_size($total) . "\t" .
      format_size($used) . "\t" .
      format_size($avail) . "\t" .
      $usePct . "\t" .
      $dirname . "\t" .
      $fs->sys_type_name . "/" . $fs->type_name . "\n";

}

sub format_size {
    my($size) = @_;
    return Sigar::format_size($size * 1024);
}
